### PR TITLE
RavenDB-13017: Database Loading Logging - Add Information + Add Stora…

### PR DESCRIPTION
**Pull request by aviviadi**
…ge.SkipChecksumOnRecovery Config Option

### CODE REVIEW

File: src/Raven.Server/Documents/DocumentsStorage.cs [hunk 1]
```diff
@@ -238,6 +238,7 @@ public void Initialize(bool generateNewDatabaseId = false)
             options.PrefetchResetThreshold = DocumentDatabase.Configuration.Storage.PrefetchResetThreshold.GetValue(SizeUnit.Bytes);
             options.SyncJournalsCountThreshold = DocumentDatabase.Configuration.Storage.SyncJournalsCountThreshold;
             options.IgnoreInvalidJournalErrors = DocumentDatabase.Configuration.Storage.IgnoreInvalidJournalErrors;
+            options.SkipChecksumValidationOnDatabaseLoading = DocumentDatabase.Configuration.Storage.SkipChecksumValidationOnDatabaseLoading;
```
Comment (ppekrol) [inline src/Raven.Server/Documents/DocumentsStorage.cs:241]: We need to add this to all the storages that we have (configuration, indexes, etc)

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 2]
```diff
@@ -249,26 +252,50 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
...
+                else
+                {
+                    if (RuntimeInformation.OSArchitecture == Architecture.Arm || RuntimeInformation.OSArchitecture == Architecture.Arm64)
+                    {
+                        addToInitLog?.Invoke($"SkipChecksumValidationOnDbLoading set to true. Skipping checksum validation of {sortedPages.Length} pages.");
+                    }
+                    else
+                    {
+                        throw new InvalidDataException( // RavenDB-13017
+                            $"Storage.SkipChecksumValidationOnDatabaseLoading set to true is not allowed on non ARM architecture. This instance running on {RuntimeInformation.OSArchitecture}");
```
Comment (arekpalinski) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs]: minor: `Storage.` -> `Options.` ? maybe use `nameof` here

File: src/Raven.Server/Config/Categories/StorageConfiguration.cs [hunk 3]
```diff
@@ -92,5 +92,10 @@ public class StorageConfiguration : ConfigurationCategory
         [DefaultValue(null)]
         [ConfigurationEntry("Storage.Dangerous.IgnoreInvalidJournalErrors", ConfigurationEntryScope.ServerWideOnly)]
         public bool? IgnoreInvalidJournalErrors { get; set; }
+
+        [Description("EXPERT: Skip checksum validation on database loading process (applicable only for ARM 32/64)")]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/StorageConfiguration.cs:96]: Why applicable to ARM only? Don't we want it to be generally available?

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 4]
```diff
@@ -249,26 +252,50 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
...
+                    // we need to iterate from the end in order to filter out pages that was overwritten by later transaction
+                    addToInitLog?.Invoke($"Validate checksum on {sortedPages.Length} pages");
+                    var sp = Stopwatch.StartNew();
+                    for (var i = sortedPages.Length - 1; i >= 0; i--)
                     {
-                        var modifiedPage = sortedPages[i];
+                        if (sp.Elapsed.TotalSeconds >= 60)
+                        {
+                            sp.Restart();
+                            addToInitLog?.Invoke($"Still calculating checksum.. ({sortedPages.Length - i} out of {sortedPages.Length}");
```
Comment (ayende) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs]:
```suggestion
                            addToInitLog?.Invoke($"Still calculating checksum... ({sortedPages.Length - i} out of {sortedPages.Length}");
```

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 5]
```diff
@@ -249,26 +252,50 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
...
 
-                        _env.ValidatePageChecksum(modifiedPage, ptr);
+                            _env.ValidatePageChecksum(modifiedPage, ptr);
 
-                        minPageChecked = modifiedPage;
+                            minPageChecked = modifiedPage;
+                        }
+                    }
+
+                    addToInitLog?.Invoke($"Validate completed");
```
Comment (ayende) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs]: Add the number of pages and the time that it took

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 6]
```diff
@@ -249,26 +252,50 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
...
+                }
+                else
+                {
+                    if (RuntimeInformation.OSArchitecture == Architecture.Arm || RuntimeInformation.OSArchitecture == Architecture.Arm64)
```
Comment (ayende) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs:291]: Why do we have this limitation? If we are running on low end x86 machine, we might want this as well

File: src/Voron/Impl/Journal/WriteAheadJournal.cs [hunk 7]
```diff
@@ -304,6 +331,7 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
 
             _journalIndex = lastFlushedJournal;
 
+            addToInitLog?.Invoke($"Cleanup Newer Invalid Journal Files");
```
Comment (ayende) [inline src/Voron/Impl/Journal/WriteAheadJournal.cs]: Add `lastFlushedJournal` here

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Config/Categories/StorageConfiguration.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Server/Config/Categories/StorageConfiguration.cs:96]: On my request.

Comment (ayende) [reply src/Raven.Server/Config/Categories/StorageConfiguration.cs:96]: What is the reasoning?

**Pull request merged**